### PR TITLE
LPS-177939 know if a given user agent is a crawler user agent (via a configuration list) and act according to that

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
to activate : FF : LPS-175850

On the second last commit I'm making not visible the configuration due to the story of the configuration (that will merge and need to be refined is https://issues.liferay.com/browse/LPS-180049

- LPS-177939 generate new configuration for the crawler user agents.
- LPS-177939 generate CrawlerUserAgentsProvider that will be used to obtain the crawler user agents for the other services
- LPS-177939 use the new provider to get the crawler user agents that now is a set strings
- LPS-177939 when updating the configuration the returned crawlerUserAgents are processed
- LPS-177939 create a method to know if a given user agent is a crawler user agent
- LPS-177939 fix test
- LPS-177939 no longer needed
- LPS-177939 add test to check the new provider
- LPS-177939 buildLang
- LPS-177939 hide the configuration till the story is made
- LPS-177939 baseline

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/internal/provider/CrawlerUserAgentsProviderImpl.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		}
+
+		return false;
+	}
+
+	@Override
+	public void updated(Dictionary<String, ?> dictionary)
+		throws ConfigurationException {
+
+		_crawlerUserAgents = _getCrawlerUserAgents();
```
Comment (adolfopa) [inline modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/internal/provider/CrawlerUserAgentsProviderImpl.java:71]: This looks strange. Shouldn't you update the `_crawlerUserAgents` based on the dictionary contents? Also, having the updated event hook doesn't make the activate method redundant?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/internal/provider/CrawlerUserAgentsProviderImpl.java (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/internal/provider/CrawlerUserAgentsProviderImpl.java:71]: > Also, having the updated event hook doesn't make the activate method redundant?

This is redundant, but the `updated` it's only called when the configuration is **saved** so if you let the configuration as default and you do not touch it, the `_crawlerUserAgents` is never initialized so is `null`, so the activate, in my opinion is needed.

So, the `updated` is needed too (by the `managedService`)

@adolfopa 
What do you think of removing the managedService and add the `@Modified `on the `activate` method?

Comment (adolfopa) [reply modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/internal/provider/CrawlerUserAgentsProviderImpl.java:71]: Sounds good.

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Pull request closed without merging**